tools/mmp/aot.cs

```diff
@@ -214,18 +216,65 @@ public void Compile (IFileEnumerator files)
 				throw ErrorHelper.CreateError (0099, Errors.MX0099, $"\"AOTBundle with aot: {options.CompilationType}\" ");
 
 			var monoEnv = new Dictionary<string, string> { { "MONO_PATH", files.RootDir } };
-
 			List<string> filesToAOT = GetFilesToAOT (files);
-			Parallel.ForEach (filesToAOT, ParallelOptions, file => {
+
+			bool needsLipo = abis.Length > 1;
+			string tempAotDir = Path.GetDirectoryName (filesToAOT [0]);
+			if (needsLipo) {
+				foreach (var abi in abis) {
+					Directory.CreateDirectory (Path.Combine(tempAotDir, abi.AsArchString()));
+				}
+			}
+
+			Parallel.ForEach (filesToAOT.SelectMany(f => abis, (file, abi) => new Tuple<string, Abi>(file, abi)), ParallelOptions, tuple => {
+				var file = tuple.Item1;
+				var abi = tuple.Item2;
+
 				var cmd = new List<string> ();
-				cmd.Add (options.IsHybridAOT ? "--aot=hybrid" : "--aot");
+				var aotArgs = new List<string> ();
+				aotArgs.Add ($"mtriple={abi.AsArchString()}");
+				if (options.IsHybridAOT)
+					aotArgs.Add ("hybrid");
+				if (needsLipo)
+					aotArgs.Add ($"outfile={Path.Combine(tempAotDir, abi.AsArchString(), Path.GetFileName (file) + ".dylib")}");
+				cmd.Add ($"--aot={string.Join(",", aotArgs)}");
 				if (IsModern)
 					cmd.Add ("--runtime=mobile");
 				cmd.Add (file);
-				if (RunCommand (MonoPath, cmd, monoEnv) != 0)
+				if (RunCommand (GetMonoPath(abi), cmd, monoEnv) != 0)
 					throw ErrorHelper.CreateError (3001, Errors.MX3001, "AOT", file);
 			});
 
+			// Lipo the result
+			if(needsLipo) {
+				Parallel.ForEach (filesToAOT, ParallelOptions, file => {
+					string [] inputs = abis.Select (abi => Path.Combine (tempAotDir, abi.AsArchString (), Path.GetFileName (file) + ".dylib")).ToArray();
+					string output = file + ".dylib";
+
+					Driver.RunLipo (Driver.App, output, inputs);
+					foreach(var input in inputs) {
+						File.Delete (input);
+					}
+
+					string [] dsymInputDirs = inputs.Select (input => $"{input}.dSYM").ToArray ();
```

Comment:
Yes, `dsymutil` works on fat executables. My mistake again : I thought that the AOT compiler stripped the symbols afterward from the output, but it actually doesn't. Xamarin.Mac doesn't seem to strip those executables (I actually do on my own projects with an extra MSBuild task). I'll change it to use `dsymutil` instead, it should simplify that code quite a bit.

Reply:
However, I just noticed one mistake in my code : the main executable doesn't have a dSYM anymore : because we generate all archs in another folder, and lipo the executable together, the dSYM is never added! However, from what I see, there is no other choice but to lipo the dSYM together on that part (the symbols aren't in the executable, only in the dSYM, once clang has run, and dsymutil only takes one executable to generate the dSYM). I'll implement that and check to make sure it still works.

Reply:
> Yes, `dsymutil` works on fat executables. My mistake again : I thought that the AOT compiler stripped the symbols afterward from the output, but it actually doesn't. Xamarin.Mac doesn't seem to strip those executables (I actually do on my own projects with an extra MSBuild task). I'll change it to use `dsymutil` instead, it should simplify that code quite a bit.

And turns out I was wrong on this one as well, I tried with `CreateDsym` from the content of the lipo-ed file and turns out there is no debug section in those files, it's all in the dsym. Since I already need that logic for the main executable dSYM, I'll write a common function for merging multiple abi dsyms and use that in both locations

Reply:
Actually wait, this should not be needed at all, because we should run dsymutil on the actual .app from the msbuild tasks after mmp is done (if we're not doing it that way already, we should, because that's how we do it for iOS).

Reply:
We can't run dsymutil on an executable that doesn't contain symbols. I just gave it a try and it just complains that there is no symbol, yielding no resulting dsym. On iOS you're building one big executable which has the symbols at that moment, but that's not the case on Xamarin.Mac.

Reply:
I implemented what I just suggested (common function to merge dSYM for multiple abis) at f85d54097e4ff7048278ca1b0dd803bec16da431, tested on my end (I have some extra MSBuild on my end to grab every dSYM in the resulting app and package them in one big dSYM because Xamarin.Mac doesn't support it) and every dSYM is available properly and contains both architectures